SteamKit2/SteamKit2/Steam/CDN/Client.cs

```diff
@@ -83,9 +84,19 @@ public async Task<DepotManifest> DownloadManifestAsync( uint depotId, ulong mani
 
             var manifestData = await DoRawCommandAsync( server, url, proxyServer ).ConfigureAwait( false );
 
-            manifestData = ZipUtil.Decompress( manifestData );
+            // Decompress the zipped manifest data
+            using var ms = new MemoryStream( manifestData );
+            using var zip = new ZipArchive( ms );
+            var entries = zip.Entries;
 
-            var depotManifest = new DepotManifest( manifestData );
+            DebugLog.Assert( entries.Count == 1, nameof( ZipUtil ), "Expected the zip to contain only one file" );
+
+            DepotManifest depotManifest;
+
+            using ( var zipEntryStream = entries[ 0 ].Open() )
+            {
+                depotManifest = DepotManifest.Deserialize( zipEntryStream );
+            }
```

Comment:
Could we simplify this into a helper function, something like:

```csharp
var depotManifest = ZipUtil.Decompress( manifestData, static zipEntryStream => DepotManifest.Deserialize( zipEntryStream ) );
```
?

Reply:
It would have to return DepotManifest from that lambda, seems wack.

Reply:
ZipUtil.Decompress<T>(..., Func<Stream, T>)

?

Reply:
I won't be able to check CRC anyway, is it really worth the couple lines? Also I'm going to push a commit that streams from http into a rented buffer, that unfortunately has some copypasta, so want your thoughts on the approach.